### PR TITLE
[HWKINVENT-88] Fix stable ID computation in REST

### PR DESCRIPTION
**Pull request by metlos**

### CODE REVIEW

File: api/src/main/java/org/hawkular/inventory/api/model/CanonicalPath.java [hunk 1]
```diff
@@ -278,27 +278,43 @@ public String getEnvironmentId() {
         }
 
         public String getMetricTypeId() {
-            return idIfTypeCorrect(getRoot().down(), MetricType.class);
+            return idIfTypeCorrect(CanonicalPath.this, Resource.class);
```
Comment (jkremser) [inline api/src/main/java/org/hawkular/inventory/api/model/CanonicalPath.java]: `s/Resource/Metric(Type)/` ?

Comment (metlos) [inline api/src/main/java/org/hawkular/inventory/api/model/CanonicalPath.java]: Oh, the joys of copy paste :) Thanks for spotting it.

File: api/src/main/java/org/hawkular/inventory/api/model/CanonicalPath.java [hunk 2]
```diff
@@ -278,27 +278,43 @@ public String getEnvironmentId() {
...
         }
 
         public String getResourceTypeId() {
-            return idIfTypeCorrect(getRoot().down(), ResourceType.class);
+            return idIfTypeCorrect(CanonicalPath.this, Resource.class);
```
Comment (jkremser) [inline api/src/main/java/org/hawkular/inventory/api/model/CanonicalPath.java]: This looks suspicious too. `getResourceTypeId` -> `ResourceType.class` (?)

File: api/src/main/java/org/hawkular/inventory/api/model/CanonicalPath.java [hunk 3]
```diff
@@ -278,27 +278,43 @@ public String getEnvironmentId() {
...
+            }
+
+            return ret.get();
         }
 
         public String getMetricId() {
-            String id = idIfTypeCorrect(getRoot().down(2), Metric.class);
-
-            return id != null ? id : idIfTypeCorrect(getRoot().down(3), Metric.class);
+            return idIfTypeCorrect(CanonicalPath.this, Resource.class);
```
Comment (jkremser) [inline api/src/main/java/org/hawkular/inventory/api/model/CanonicalPath.java]: ---||---

**Pull request merged**